src/app/locator/qgsgotolocatorfilter.cpp

```diff
@@ -87,51 +88,114 @@ void QgsGotoLocatorFilter::fetchResults( const QString &string, const QgsLocator
     }
   }
 
-  if ( okX && okY )
+  if ( okX && okY )  // Ok found a valid looking x,y coordinate pair, check against Wgs84crs, MapCanvas-crs AND layer-crs's
   {
-    QVariantMap data;
-    const QgsPointXY point( posX, posY );
-    data.insert( QStringLiteral( "point" ), point );
-
-    const bool withinWgs84 = wgs84Crs.bounds().contains( point );
-    if ( !posIsDms && currentCrs != wgs84Crs )
+    // collect all crs's from all layers (plus MapCanvas crs and wgs84)
+    QList<QString> crsList;  // TODO make this a list
+    crsList.append( wgs84Crs.authid() );
+    if ( currentCrs != wgs84Crs )
+      crsList.append( currentCrs.authid() );
+    for ( const QgsMapLayer *layer : QgsProject::instance()->mapLayers() )
+    {
+      if ( !crsList.contains( layer->crs().authid() ) )
+      {
+        crsList.append( layer->crs().authid() );
+      }
+    }
+    // Now go over all crs's and check if the coordinate COULD be use in that crs
+    for ( const QString &authid : crsList )
     {
+
+      QgsCoordinateReferenceSystem crs = QgsCoordinateReferenceSystem( authid );
+
+      QVariantMap data;
+      const QgsPointXY point( posX, posY );
+      data.insert( QStringLiteral( "point" ), point );
+
       QgsLocatorResult result;
       result.filter = this;
-      result.displayString = tr( "Go to %1 %2 (Map CRS, %3)" ).arg( locale.toString( point.x(), 'g', 10 ), locale.toString( point.y(), 'g', 10 ), currentCrs.userFriendlyIdentifier() );
-      result.userData = data;
-      result.score = 0.9;
-      emit resultFetched( result );
-    }
 
-    if ( withinWgs84 )
-    {
-      if ( currentCrs != wgs84Crs )
+      const bool withinWgs84 = wgs84Crs.bounds().contains( point );
+      if ( crs == wgs84Crs && crs == currentCrs && withinWgs84 )
       {
-        const QgsCoordinateTransform transform( wgs84Crs, currentCrs, QgsProject::instance()->transformContext() );
-        QgsPointXY transformedPoint;
-        try
+        result.displayString = tr( "Go to %1° %2° (%3)" ).arg( locale.toString( point.x(), 'g', 10 ), locale.toString( point.y(), 'g', 10 ), wgs84Crs.userFriendlyIdentifier() );
+        result.score = 1.0;
+        result.userData = data;
+        emit resultFetched( result );
+        continue;
+      }
+
+      // non wgs84 project, create a transform to be able to check point against bounds
+      const QgsCoordinateTransform transform2wgs84( crs, wgs84Crs, QgsProject::instance()->transformContext() );
```

Comment:
```suggestion
      const QgsCoordinateTransform transform2wgs84( crs, wgs84Crs, QgsProject::instance()->transformContext() );
      transform2wgs84.setBallparkTransformsAreAppropriate( true );
```

Reply:
Ah, thanks was not aware of this, so no more warnings?
(would be cool to also have something like that for that pesky "no crs defined, will use Lat Lon/WGS84 for layer" (or something like that) ... :-)